lib/internal/url.js

```diff
@@ -534,15 +533,27 @@ ObjectDefineProperties(URLSearchParams.prototype, {
   },
 });
 
+/**
+ * Checks if a value has the shape of a WHATWG URL object.
+ *
+ * Using a symbol or instanceof would not be able to recognize URL objects
+ * coming from other implementations (e.g. in Electron), so instead we are
+ * checking some well known properties for a lack of a better test.
+ *
+ * @param {*} self
+ * @returns {self is URL}
+ */
 function isURL(self) {
-  return self != null && ObjectPrototypeHasOwnProperty(self, context);
+  return self != null && self.href && self.origin;
```

Comment:
```suggestion
  return Boolean(self?.href && self.origin);
```

Reply:
Could origin be an empty string as valid value?

Reply:
No, it will always be a string. It can be `"null"` but not `null` or `undefined`. Here's the spec reference: https://html.spec.whatwg.org/multipage/browsers.html#ascii-serialisation-of-an-origin

Reply:
> Could origin be an empty string as valid value?

https://html.spec.whatwg.org/multipage/browsers.html#origin

My understanding is that the response is no, as an opaque origin would get serialized to `"null"`.

Reply:
Here's the appropriate implementation in Ada: https://github.com/ada-url/ada/blob/main/src/url-getters.cpp#L41

Reply:
Thanks. I would still suggest to change the return type to boolean.

Reply:
Is the change purely for aesthetic purposes, or would it be faster? My gut says that casting would be slower, but I wanted to learn/understand your reasoning.

Reply:
The performance should be identical (the cast should implicitly be done at the callsite, if it's compared as boolean). It's just that a `isFoobar()` method should always return a boolean and not a string. Otherwise it breaks with the principle of least surprise.

Reply:
Ah I see. You're right. I'm updating the pull request. Thank you for the review & explanation!

Reply:
Updated the pull request. Resolving. Thanks for the review!